### ISSUE TITLE
Simplify names output by `target_sda`

### ISSUE THREAD

**Issue by jdhoffa** [RMI-PACTA/r2dii.analysis#313]
In particular, the most important columns output by `target_sda` are `emission_factor_metric` and `emission_factor_value`, whereas `target_market_share` outputs the more simple columns `metric` and `production` or `technology_share`. 

I propose:
* `emission_factor_metric` -> `metric`
* `emission_factor_value` -> `emission_factor`

``` r
library(dplyr)
library(r2dii.data)
library(r2dii.match)
library(r2dii.analysis)

matched <- loanbook_demo %>% 
  match_name(ald_demo) %>% 
  prioritize()

target_market_share(
  matched, 
  ald_demo, 
  scenario_demo_2020,
  region_isos_demo
) %>% 
  names()
#> [1] "sector"           "technology"       "year"             "region"          
#> [5] "scenario_source"  "metric"           "production"       "technology_share"

target_sda(
  matched, 
  ald_demo, 
  co2_intensity_scenario_demo
) %>% 
  names()
#> Warning: Removing ald rows where `emission_factor` is NA
#> [1] "sector"                 "year"                   "emission_factor_metric"
#> [4] "emission_factor_value"
```

<sup>Created on 2021-06-17 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by maurolepore** [comment]
@AlexAxthelm, @tposey28, 

Based on your experience working with data, what would be best practice here? One of our CRAN packages has a  function `f()` that outputs the column `long_name` column and we want to shorten it to just `name`.  The package has no reverse dependencies on CRAN but users might have scripts that would break. 

What would the next version of the package do?

1. Just do it. `f()` will output `name` not `long_name`.
2. Support both temporarily. `f()` will output  both `long_name` and `name` with a warning that `long_name` will be dropped soon.
3. Opt in. `f()` will only output the original `long_name` with a deprecation warning. The user can opt in the new `name` by setting an option, e.g. `options(package.name.use_short_names = TRUE)`.

**Comment by tposey28** [comment]
Option 2 seems like the best, but also option 1 with a warning upon package load that tells the user about the change could also work.
________________________________

**Comment by AlexAxthelm** [comment]
@maurolepore Can you point me toward the source for `f()`? I'm having trouble imaging what (2) would look like.

**Comment by jdhoffa** [comment]
The function is `target_sda`. 
Compare the two outputs here: 

``` r
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
library(r2dii.data)
library(r2dii.match)
library(r2dii.analysis)

matched <- loanbook_demo %>% 
  match_name(ald_demo) %>% 
  prioritize()

out <- target_sda(
  matched, 
  ald_demo, 
  co2_intensity_scenario_demo
) 
#> Warning: Removing ald rows where `emission_factor` is NA

# current
out %>% 
  select(
    emission_factor_metric,
    emission_factor_value
  )
#> # A tibble: 239 x 2
#>    emission_factor_metric emission_factor_value
#>    <chr>                                  <dbl>
#>  1 projected                              0.658
#>  2 projected                              0.659
#>  3 projected                              0.660
#>  4 projected                              0.661
#>  5 projected                              0.662
#>  6 projected                              0.662
#>  7 projected                              0.663
#>  8 projected                              0.664
#>  9 projected                              0.665
#> 10 projected                              0.666
#> # … with 229 more rows

# proposed
out %>% 
  select(
    metric = emission_factor_metric,
    emission_factor = emission_factor_value
  )
#> # A tibble: 239 x 2
#>    metric    emission_factor
#>    <chr>               <dbl>
#>  1 projected           0.658
#>  2 projected           0.659
#>  3 projected           0.660
#>  4 projected           0.661
#>  5 projected           0.662
#>  6 projected           0.662
#>  7 projected           0.663
#>  8 projected           0.664
#>  9 projected           0.665
#> 10 projected           0.666
#> # … with 229 more rows
```

<sup>Created on 2021-06-17 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by jdhoffa** [comment]
The change is tentatively implemented in #314

**Comment by AlexAxthelm** [comment]
Okay. the specifics are making more sense now. Thanks for the context.

My big question at this point: Is making this an aesthetic issue, or is there a workflow purpose to it? Will making the `target_*()` functions consistent going to make it easier or harder for users to make mistakes? Ultimately, is thee output of `target_sda()` able to bee used as a drop-in replacement for the output of `target_market_share`?

If those two outputs are supposed to be used for the same purpose (whatever the next step of the pipeline is), then I'm totally okay with this.
